### ISSUE TITLE
Disable editing of beatmap descriptions after beatmap is ranked

### ISSUE THREAD

**Issue by peppy** [ppy/osu-web#8866]
Permissions should be similar to guest mapper assignment. It should not be editable by the mapper after rank.

**Comment by MkGuh** [comment]
I disagree. I don't know what cases of abuse ever happened from this, but instead of disallowing mappers to do that maybe you should improve the tools available for staff members to monitor description changes made in Ranked beatmaps by the mapper.

**Comment by peppy** [comment]
Ah, this was not intended to be a discussion sorry, this is a work task and required from my end.

We have pinned comments these days which can be used to provide updates to maps (mappers can use these already).

**Comment by Dada-jodel** [comment]
Surely the convenience and uses of editing a description while the map is ranked (updating art, tracking previous and future ranked maps in order, giving props to good plays, changing GD info such as names or icons when SR is updated) outweighs the extremely rare cases of abuse of ranked map descriptions which are already dealt with as soon as they are found by GMTs / NATs.

I see no reason to dismiss a discussion around this, given that anyone who's remotely familiar with the system as it stands figures it's a change that would make the system in question actively worse. Pinned comments are not looked at in the same way as descriptions as they are not visible upon opening the page unless scrolled to - and if the intention here is that if someone has to ask a moderator to change their description for them every time they wish to do so, then surely the burden of doing that is much, much greater than removing abuse cases (of which there were only a very limited amount). An analogous case to this would be something like having to parse every beatmap submission through a moderator to avoid the issue that pops up much more often, in comparison, of racist or political or otherwise innapropriate submissions being published. We can both agree that this isn't practical or, y'know, sensical in the slightest.

Please do not dismiss discussion when making a change like this in the future as it's bound to lead to cases like this - where pretty much everyone involved with this system immediately saw how bad this change was when it was brought up.

**Comment by peppy** [comment]
In osu!, we aim for preserving all content. Allowing content to be edited after in a final state goes against this. I understand your opinions on the matter, but disagree with them from a fundamental perspective of how the ranking system and game works.

We've had cases of
- People replacing their ranked beatmap description with nothing, and having it lost forever (if they decide to quit or destroy content for whatever reason)
- People linking off-site downloads which contain differing content (which is generally not condoned in that description field)
- People bypassing DMCA takedowns and causing further headaches.

Your listed use cases of "updating mapper names" can be fixed by using user ID links (the standrad these days). Previous names are shown on profiles and the links will still work.

Your use case of changing icons when SR changes seems a bit out of scope. I don't think you should be using inline icons based on SR since it is a constantly changing number. You should base it on the difficulty naming/brackets instead.

Your use case of updating art seems weird. This should be finalised before applying to have your beatmap ranked (since it should be in a final state at that point).

Your use case of updating "previous and future ranked maps" is valid I guess, but can be done via asking a team member to update the link on your behalf, or linking to your profile instead.

Just like you can't update the *content* of a ranked beatmap, you should not be able to update the description. It is finalised metadata in the eyes of the game systems.

**Comment by mm201** [comment]
A few more use cases:

* Crediting the map nominators
* Fixing dead links to images that were hosted on file locker services that changed their retention policy or for which the mapper stopped paying
* Updating the descriptions of old maps (often just a meme or a single lyric) to match your current standards, or to match current guidelines re: crediting asset artists

The BNs will also need to be made aware of this change, that map descriptions are now locked-in content that will need to be checked before nomination.

> but can be done via asking a team member to update the link on your behalf, or linking to your profile instead.

It seems the concern with this is that tracking down a staff member for possibly minor changes is overkill. It will increase workload, take time, and create a chilling effect on the maintenance of map descriptions.

**Comment by MkGuh** [comment]
If a beatmap description shouldn't be edited after it is Ranked because it's against the principles of the game, then why are we allowed to ask for moderators to edit stuff for us? As mm201 said, it'd only increase workload.

I still think you should improve the tools given to staff members to easily check edits made to a map's description after it is Ranked than to block mappers from editing it completely because of a few cases of abuse or because it is against the principles of Ranked content of the game.

**Comment by peppy** [comment]
> It seems the concern with this is that tracking down a staff member for possibly minor changes is overkill. It will increase workload, take time, and create a chilling effect on the maintenance of map descriptions.


It's an understandable concern that as I hope you realise, I can understand that much when making this issue. But the current method of allowing updates by anyone at and point in time with no edit history or ability to recover from abusive/unwanted edits is an oversight which needs to be addressed.

Also I'm not sure the throughput of updates to descriptions is, will check on that before applying any changes and make sure the NAT are on board to handle that throughput.

**Comment by peppy** [comment]
Locking this thread for now as it looks like it has been shared in a larger scope than expected. No changes will be made unless the NAT is on board.
